### PR TITLE
feat(dashboard): Transition to Explore with React Router

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY

Currently, when user clicks on chart title or "Edit chart" button in Dashboard view, we first make a POST request to get form_data_key and then open Explore in a new window. In order to fully benefit from Explore SPA project, we need to stay in the SPA context when transitioning from Dashboard to Explore - which means that we should open Explore in the same tab.

This PR changes the default behaviour to open Explore in the same tab using the React Router. If user wants to open Explore in separate tab, they can do cmd+click (on Mac) or ctrl+click (on Linux/Windows). The tooltip was adjusted to make the feature easier to discover. 

Why can't we just make the chart title a link and let user "right click -> open in a new tab"? When user clicks "Edit chart", we first need to send an API request to get the form_data_key, which is required to correctly pass dashboard context to Explore.   For this reason we can't use standard HTML <a> element and instead we use a button with `onClick` callback.
In the future, we'll work on improving this behaviour, so that we can use a link to transition to Explore.

Since this change in flow (open Explore in the same tab by default) might be disruptive, user can bring back the old behaviour by enabling the new feature flag `DASHBOARD_EDIT_CHART_IN_NEW_TAB`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/177313748-4b63608e-d761-4ef1-904c-bb4733d73648.mov

### TESTING INSTRUCTIONS

1. Go to dashboard
2. Click on chart title or "Edit chart" in menu or "Edit chart" in "View as table" modal
3. Verify that Explore opens in the same tab
4. Go back to Dashboard
5. Click on chart title while holding CMD
6. Verify that Explore opens in new tab

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by ktmud** [comment]
Flagging https://github.com/apache/superset/issues/20381 as this is related. I agree we should use a regular link for transition. The form data retrieval (and subsequent redirects) should happen on the Explore page, not dashboard page.

**Comment by kgabryje** [comment]
> Flagging #20381 as this is related. The form data retrieval (and subsequent redirects) should happen on the Explore page, not dashboard page.

I think #20381 is related to permissions, this PR is part of the Explore SPA project.

> The form data retrieval (and subsequent redirects) should happen on the Explore page, not dashboard page.

Form data retrieval does happen on Explore. What we do on dashboard is post form data with native filters, color scheme and other data related to dashboard and retrieve a form_data_key. We need to figure out a way to skip that step in order to be able to use a link.

Since this change in flow (open Explore in the same tab by default) might be disruptive, I put it behind a new feature flag `DASHBOARD_TO_EXPLORE_SPA`, which is enabled by default.

**Comment by ktmud** [comment]
I misspoke. What I meant is the form data key should be created on the Explore page (and in the Explore page's route if using SPA navigation). Context in the dashboard page should be carried over to Explore via URL params by default and only very large payload will use localStorage if needed (although I doubt it will happen very often).


It's not just permissions, but how making a link depends on a post request has also made it less accessible. The post request adds a sometimes significant delay between users clicking the link to the page opening, which often confuses the users.

File: superset/config.py [hunk 1]
```diff
@@ -429,6 +429,7 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
     "GENERIC_CHART_AXES": False,
     "ALLOW_ADHOC_SUBQUERY": False,
     "USE_ANALAGOUS_COLORS": True,
+    "DASHBOARD_EDIT_CHART_IN_TAB": True,
```
Comment (michael-s-molina) [inline superset/config.py:432]: I can understand the confusion here 😆. I meant `DASHBOARD_EDIT_CHART_IN_NEW_TAB` (added NEW) and this should be `false` by default.

Comment (kgabryje) [reply superset/config.py:432]: And I understood that as `edit in SAME tab` 😆

Comment (michael-s-molina) [reply superset/config.py:432]: My bad 😆

Comment (kgabryje) [reply superset/config.py]: Updated, also updated PR description

**Review by michael-s-molina — APPROVED** [review]
LGTM. I tested with the feature flag enabled and disabled.

Non-blocking suggestion: also add the tooltip to the Edit Chart option in the 3 dots menu.

**Comment by michael-s-molina** [comment]
Can you also add something to `UPDATING.md`?

**Review by zhaoyongjie — APPROVED** [review]

**Pull request merged**